server/service/client_queries.go

```diff
@@ -13,143 +10,30 @@ import (
 	verb, path := "GET", "/api/v1/fleet/spec/queries/"+url.PathEscape(name)
-	response, err := c.AuthenticatedDo(verb, path, "", nil)
-	if err != nil {
-		return nil, fmt.Errorf("%s %s: %w", verb, path, err)
-	}
-	defer response.Body.Close()
-
-	switch response.StatusCode {
-	case http.StatusNotFound:
-		return nil, notFoundErr{}
-	}
-	if response.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf(
-			"get query received status %d %s",
-			response.StatusCode,
-			extractServerErrorText(response.Body),
-		)
-	}
-
 	var responseBody getQuerySpecResponse
-	err = json.NewDecoder(response.Body).Decode(&responseBody)
-	if err != nil {
-		return nil, fmt.Errorf("decode get query spec response: %w", err)
-	}
-
-	if responseBody.Err != nil {
-		return nil, fmt.Errorf("get query spec: %s", responseBody.Err)
-	}
-
-	return responseBody.Spec, nil
+	err := c.authenticatedRequest(nil, verb, path, &responseBody)
+	return responseBody.Spec, err
 }
 
 // GetQueries retrieves the list of all Queries.
 func (c *Client) GetQueries() ([]*fleet.QuerySpec, error) {
-	response, err := c.AuthenticatedDo("GET", "/api/v1/fleet/spec/queries", "", nil)
-	if err != nil {
-		return nil, fmt.Errorf("GET /api/v1/fleet/spec/queries: %w", err)
-	}
-	defer response.Body.Close()
-
-	if response.StatusCode != http.StatusOK {
-		return nil, fmt.Errorf(
-			"get queries received status %d %s",
-			response.StatusCode,
-			extractServerErrorText(response.Body),
-		)
-	}
-
+	verb, path := "GET", "/api/v1/fleet/spec/queries"
 	var responseBody getQuerySpecsResponse
-	err = json.NewDecoder(response.Body).Decode(&responseBody)
-	if err != nil {
-		return nil, fmt.Errorf("decode get query spec response: %w", err)
-	}
-
-	if responseBody.Err != nil {
-		return nil, fmt.Errorf("get query spec: %s", responseBody.Err)
-	}
-
-	return responseBody.Specs, nil
+	err := c.authenticatedRequest(nil, verb, path, &responseBody)
+	return responseBody.Specs, err
 }
 
 // DeleteQuery deletes the query with the matching name.
 func (c *Client) DeleteQuery(name string) error {
 	verb, path := "DELETE", "/api/v1/fleet/queries/"+url.PathEscape(name)
-	response, err := c.AuthenticatedDo(verb, path, "", nil)
-	if err != nil {
-		return fmt.Errorf("%s %s: %w", verb, path, err)
-	}
-	defer response.Body.Close()
-
-	switch response.StatusCode {
-	case http.StatusNotFound:
-		return notFoundErr{}
-	}
-	if response.StatusCode != http.StatusOK {
-		return fmt.Errorf(
-			"delete query received status %d %s",
-			response.StatusCode,
-			extractServerErrorText(response.Body),
-		)
-	}
-
 	var responseBody deleteQueryResponse
-	err = json.NewDecoder(response.Body).Decode(&responseBody)
-	if err != nil {
-		return fmt.Errorf("decode get query spec response: %w", err)
-	}
-
-	if responseBody.Err != nil {
-		return fmt.Errorf("get query spec: %s", responseBody.Err)
-	}
-
-	return nil
-}
-
-func (c *Client) CreateQuery(name, query, description string) (*fleet.Query, error) {
```

Comment:
Same here, `CreateQuery` is completely removed.

Reply:
Yup, same logic as before.